libs/ktor2client-onbehalfof/src/main/kotlin/DownstreamResourceClass.kt

```diff
@@ -23,13 +24,39 @@ class DownstreamResourceClient(
     private val azureAdClient: AzureAdClient,
     private val httpClient: HttpClient = defaultHttpClient
 ) {
+    suspend fun get(
+        resource: Resource,
+        accessToken: AccessTokenWrapper
+    ): Result<Resource, ThrowableErrorMessage> {
+        val scopes = listOf("api://${resource.clientId}/.default")
+        return hentTokenFraAD(accessToken, scopes)
+            .andThen { oboAccessToken ->
+                fetchFromDownstreamApi(resource, oboAccessToken)
+            }
+            .andThen { response ->
+                Ok(resource.addResponse(response))
+            }
+    }
+
+    private suspend fun hentTokenFraAD(
+        accessToken: AccessTokenWrapper,
+        scopes: List<String>
+    ): Result<AccessToken, ThrowableErrorMessage> = if (accessToken.oid == accessToken.sub) {
```

Comment:
kanskje hade kunnet trekke ut denne sjekken for å si hva det er, d.v.s noe a.la `erMaskinTilMaskin(..)`?